src/static/js/modules/jquery/cf_formValidator.js

```diff
@@ -0,0 +1,185 @@
+        var notCheckbox = ( type === 'checkgroup' && $input.attr( 'type' ) !== 'checkbox' );
+
+        if ( notRequired || notEmail || notRadio || notCheckbox ) {
+          status[type] = null;
+          return;
+        }
+
+        if ( type === 'required' ) {
+          validation = validate.single( value, { presence: true } );
+        } else if ( type === 'email' ) {
+          validation = validate.single( value, { email: true } );
+        }
+        status[type] = ( validation === undefined );
+      } );
+
+      return status;
+    },
+
+    // Validate the fields of our form
+    // @return {object} Two inner objects containing the valid and invalid fields
+    _validateFields: function() {
+      var checkgroups = {};
+      validator.validatedFields = {
+        valid:    [],
+        invalid:  []
+      };
+
+      validator.fields.each( function( i ) {
+        var $input  = $( this );
+        var failed  = false;
+        var field;
+
+        // if it's a button elem or type is button or submit, skip it
+        if ( $input.is( 'button, :button, :submit' ) ) {
+          return;
+        // if it's a checkbox test the entire check group
+        } else if ( $input.is( ':checkbox' ) ) {
+          var name    = $input.attr( 'name' );
+          var $group  = $input.closest( '.form-group' );
+
+          if ( checkgroups[name] || !$group.hasClass( 'required-check-group' ) ) {
+            return;
+          }
+
+          var isChecked = $group.find( 'input:checked' ).length > 0;
+
+          field = {
+            elem:   $group,
+            value:  null,
+            label:  $.trim( $group.find( '.form-label-header' ).text() ),
+            status: {
+              checkgroup: isChecked
+            }
+          }
+
+          checkgroups[name] = true;
+        } else {
+          var value       = $input.val();
+          var validation  = validator._validateInput( $input, value );
+
+          field = {
+            elem:   $input,
+            value:  value,
+            label:  $.trim( $( 'label[for="' + $input.attr( 'id' ) + '"]' ).text() ),
+            status: validation
+          };
+        }
+
+        for ( var prop in field.status ) {
+          if ( field.status[prop] === false ) {
+            failed = true;
+          }
+        }
+
+        if ( failed ) {
+          validator.validatedFields.invalid.push( field );
+        } else {
+          validator.validatedFields.valid.push( field );
+        }
+      } );
+
+      return validator.validatedFields;
+    },
+
+    // Listen for cf_formValidator submit event
+    _init_submit_listener: function() {
+      validator.elem.on( 'submit.cf_formValidator', function( event ) {
+        var $submitted = $( this );
+        validator.fields = $submitted.find( validator.settings.inputs.toString() );
+        var validatedFields = validator._validateFields();
+
+        if ( validatedFields.invalid.length ) {
+          if ( validator.settings.onFailure ) {
```

Comment:
`if` statement can be combined with prior `if` statement on L:132 with an `&&`, e.g. `if ( validatedFields.invalid.length && validator.settings.onFailure ) {`